### PR TITLE
Normalization code

### PR DESCRIPTION
**Pull request by khrynczenko**
Normalization for images. Code provided by the authors of "Classification of breast cancer histology images using Convolutional Neural Networks" with slight changes.

### CODE REVIEW

File: normalization.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+"""
+
+from __future__ import division
+import numpy as np
+
+
+def remove_invalid_vectors(optical_density_matrix: np.ndarray,
+                           beta: float) -> np.ndarray:
+    """
+    Returns new array of only optical densities that are
```
Comment (mmyller) [inline normalization.py]: ....that are _beautiful_?

**Review by mmyller — APPROVED** [review]

File: normalization.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    Normalizes the image.
+
+    :param image_matrix: image matrix where each cell represents vector of
+    RGB values, these values should be of type float
+    :param alpha: upper threshold for correct optical density
+    :param beta: lower threshold for correct optical density
+    :return: normalized image
+    """
+
+    Io = 240
```
Comment (szympie) [inline normalization.py:42]: Do you know a better name for that?

File: normalization.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    Y = np.transpose(np.reshape(optical_density, (h * w, c)))
+
+    C = np.linalg.lstsq(HE, Y)
+    maxC = np.percentile(C[0], 99, axis=1)
+
+    C = C[0] / maxC[:, None]
+    C = C * maxCRef[:, None]
+    Inorm = Io * np.exp(- np.dot(HERef, C))
+    Inorm = np.reshape(np.transpose(Inorm), (h, w, c), order='F')
+    Inorm = np.clip(Inorm, 0, 255)
```
Comment (szympie) [inline normalization.py]: Can 0 and 255 be substituted by a constant?

File: normalization.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    if vMin[0] > vMax[0]:
+        HE = np.array([vMin, vMax])
+    else:
+        HE = np.array([vMax, vMin])
+
+    HE = np.transpose(HE)
+    Y = np.transpose(np.reshape(optical_density, (h * w, c)))
+
+    C = np.linalg.lstsq(HE, Y)
+    maxC = np.percentile(C[0], 99, axis=1)
```
Comment (szympie) [inline normalization.py]: Can 99 be substituted by a constant?

File: normalization.py (hunk 3 above)
Comment (khrynczenko) [reply normalization.py]: fixed

File: normalization.py (hunk 4 above)
Comment (khrynczenko) [reply normalization.py]: I think so.

File: normalization.py (hunk 2 above)
Comment (khrynczenko) [reply normalization.py:42]: Nope

**Pull request merged**